FirebaseInstallations/Source/Library/FIRInstallations.m

```diff
@@ -101,6 +102,7 @@ - (instancetype)initWithAppOptions:(FIROptions *)appOptions
                  prefetchAuthToken:(BOOL)prefetchAuthToken {
   self = [super init];
   if (self) {
+    [[self class] validateAppOptions:appOptions appName:appName];
```

Comment:
For Android, we realized that we break 648 TapPresubmit tests if we add this check to the initialization of the FIS instance.
If we add this check as first line/first command of methods #getId and #getToken, we only break 82 TapPresubmit tests.
I don't know how the situation is on iOS, but this is something to consider.

Reply:
OK for iOS (chatted offline)